lib/husky_musher/app.py

```diff
@@ -151,121 +65,75 @@ def main():
-
-    >>> extract_affiliation({"unscoped-affiliation": "member;affiliate;alum"})
-    {'affiliation': 'other', 'affiliation_other': 'affiliate;alum'}
-
-    >>> extract_affiliation({"unscoped-affiliation": "member"})
-    {'affiliation': '', 'affiliation_other': ''}
-
-    >>> extract_affiliation({})
-    {'affiliation': '', 'affiliation_other': ''}
-    """
-    raw_affilations = environ.get("unscoped-affiliation", "")
-
-    # "Member" is uninteresting and uninformative; a generic catch-all.
-    # The empty string might arise from our fallback above.
-    affiliations = set(raw_affilations.split(";")) - {"member",""}
-
-    rules = [
-        ("student"  in affiliations,    {"affiliation": "student",  "affiliation_other": ""}),
-        ("faculty"  in affiliations,    {"affiliation": "faculty",  "affiliation_other": ""}),
-        ("staff"    in affiliations,    {"affiliation": "staff",    "affiliation_other": ""}),
-        ("employee" in affiliations,    {"affiliation": "staff",    "affiliation_other": ""}),
-        (len(affiliations) > 0,         {"affiliation": "other",    "affiliation_other": ";".join(sorted(affiliations))}),
-        (True,                          {"affiliation": "",         "affiliation_other": ""})]
-
-    return next(result for condition, result in rules if condition)
+    netid = request.form['netid'].lower().strip()
+
+    if not re.match(r'^[a-z][a-z0-9]{,7}$', netid):
+        raise InvalidNetId(netid)
+
+    redcap_record = fetch_participant({ 'netid': netid })
+
+    # Check if PT is already reigstered
+    registration_complete = redcap_registration_complete(redcap_record)
+    if not registration_complete:
+        # Give PT info on how to register
+        return render_template('registration_required.html', netid=netid,
+            redcap_record_exists=redcap_record is not None,
+            registration_complete=registration_complete)
+
+    # Fetch all encounter events in the past 7 days.
+    recent_encounters = fetch_encounter_events_past_week(redcap_record)
+
+    # Track noteworthy instances used in survey generation logic
+    instances: Dict[str, int] = dict()
+
+    # Look for most recent TD with testing_trigger = 'Yes'
+    instances['target'] = max_instance_testing_triggered(recent_encounters)
+    # Check if TOS exists and is marked complete on or after this instance.
+    instances['complete_tos'] = max_instance('test_order_survey', recent_encounters,
+        since=instances['target'])
+    # Check for KRs on or after this instance.
+    instances['complete_kr'] = max_instance('kiosk_registration_4c7f', recent_encounters,
+        since=instances['target'])
+    instances['incomplete_kr'] = max_instance('kiosk_registration_4c7f', recent_encounters,
+        since=instances['target'], complete=False)
+
+    if instances['complete_tos'] == get_todays_repeat_instance():
+        # We won't test this PT twice in one day
+        return render_template('test_already_ordered.html', netid=netid)
```

Comment:
Just want to confirm if this is indeed a hard and fast rule? (Two kiosk tests in one day would seem to deviate from normal flow- e.g. even inconclusives dont seem to retest on the same day, but I don't necessarily want to build logic into our applications if its not a definitive rule in the study)

Reply:
This instrument is only used for mail-ordered tests. This was a requirement I gathered [in this Slack thread](https://seattle-flu-study.slack.com/archives/C018V0WGGJ0/p1601920897046600?thread_ts=1601681690.026400&cid=C018V0WGGJ0) after talking with the kiosk and comms teams. But I can ask again just to make sure. 

It's worth nothing that a participant can only (by REDCap design) receive one Kiosk test for the same instance. If we create a new KR for them today and test them, if they come back later, we'll just re-open the KR for today's instance.

Reply:
Sounds good. Thanks Kairsten for explicitly confirming!